### PR TITLE
APP-86 Remove unnecessary source file from  gpiocontrol demo esp8266

### PR DESCRIPTION
**Pull request by Dubland**
/cc @rasendubi @forGGe

### CODE REVIEW

File: gpiocontrol/source/esp8266/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,18 @@ enable_language(C)
 
 find_package (OpenSSL REQUIRED)
 
+## where is the target environment
+set(CMAKE_FIND_ROOT_PATH /usr/arm-none-eabi)
```
Comment (rasendubi) [inline gpiocontrol/source/esp8266/CMakeLists.txt]: That looks wrong. Why `/usr/arm-none-eabi`?

File: gpiocontrol/source/esp8266/CMakeLists.txt [hunk 2]
```diff
@@ -45,45 +57,39 @@ set(ESP8266_LIBDIR "-L${CMAKE_CURRENT_SOURCE_DIR}/build -L${ESP_TOOLCHAIN_DIR}/.
                     
 set(ESP8266_LIBDIR "${ESP8266_LIBDIR} -L${ESP_SDK_BASE}/lib -L${CMAKE-CMAKE_CURRENT_SOURCE_DIR} -L${ESP_TOOLCHAIN_DIR}/ld")
 
-set(ESP8266_LDFLAGS "${ESP8266_LIBDIR} -T${ESP8266_LINKSCRIPT} -nostdlib -u call_user_start -static --start-group ${ESP8266_LIBS} -lesp-platform -lkaac_s -l${APP_NAME}_s --end-group --gc-sections -Map map")
+set(ESP8266_LDFLAGS "${ESP8266_LIBDIR} -T${ESP8266_LINKSCRIPT} -nostdlib -u call_user_start -static --start-group ${ESP8266_LIBS} -lesp-platform -lkaac_s --end-group --gc-sections -Map map")
```
Comment (rasendubi) [inline gpiocontrol/source/esp8266/CMakeLists.txt:57]: Link libraries shouldn't be set there. Use `target_link_libraries` instead (indeed, they already linked that way)

File: gpiocontrol/source/esp8266/CMakeLists.txt [hunk 3]
```diff
@@ -45,45 +57,39 @@ set(ESP8266_LIBDIR "-L${CMAKE_CURRENT_SOURCE_DIR}/build -L${ESP_TOOLCHAIN_DIR}/.
...
-set(CMAKE_C_LINK_EXECUTABLE "${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ld <LINK_FLAGS> <OBJECTS>")
-
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DESP8266_PLATFORM -Os -Wpointer-arith -Wl,-EL -fno-inline-functions -nostdlib -mlongcalls -mtext-section-literals -D__ets__ -DICACHE_FLASH")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DSSID='\"${SSID}\"' -DPWD='\"${PWD}\"' -DDEMO_ACCESS_TOKEN='\"${DEMO_ACCESS_TOKEN}\"'")
+
+set(CMAKE_C_LINK_FLAGS "${ESP8266_LDFLAGS}")
+set(CMAKE_C_LINKER "${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ld")
+set(CMAKE_C_LINK_EXECUTABLE "${CMAKE_C_LINKER} <OBJECTS> <CMAKE_C_LINK_FLAGS> <LINK_LIBRARIES> -o <TARGET>")
 
 #WiFi SSID and PASSWORD
```
Comment (rasendubi) [inline gpiocontrol/source/esp8266/CMakeLists.txt]: The comment is now wrong

File: gpiocontrol/source/esp8266/CMakeLists.txt [hunk 4]
```diff
@@ -45,45 +57,39 @@ set(ESP8266_LIBDIR "-L${CMAKE_CURRENT_SOURCE_DIR}/build -L${ESP_TOOLCHAIN_DIR}/.
...
+set(ESP8266_LDFLAGS "${ESP8266_LIBDIR} -T${ESP8266_LINKSCRIPT} -nostdlib -u call_user_start -static --start-group ${ESP8266_LIBS} -lesp-platform -lkaac_s --end-group --gc-sections -Map map")
 
 set(CMAKE_AR ${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ar)
 
 set(CMAKE_RANLIB ${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ranlib)
 
-set(CMAKE_C_LINK_EXECUTABLE "${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ld <LINK_FLAGS> <OBJECTS>")
-
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DESP8266_PLATFORM -Os -Wpointer-arith -Wl,-EL -fno-inline-functions -nostdlib -mlongcalls -mtext-section-literals -D__ets__ -DICACHE_FLASH")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DSSID='\"${SSID}\"' -DPWD='\"${PWD}\"' -DDEMO_ACCESS_TOKEN='\"${DEMO_ACCESS_TOKEN}\"'")
```
Comment (rasendubi) [inline gpiocontrol/source/esp8266/CMakeLists.txt:66]: Use `list(APPEND ...` instead

File: gpiocontrol/source/esp8266/CMakeLists.txt [hunk 5]
```diff
@@ -45,45 +57,39 @@ set(ESP8266_LIBDIR "-L${CMAKE_CURRENT_SOURCE_DIR}/build -L${ESP_TOOLCHAIN_DIR}/.
...
+
+set(CMAKE_C_LINK_FLAGS "${ESP8266_LDFLAGS}")
+set(CMAKE_C_LINKER "${ESP_TOOLCHAIN_DIR}/xtensa-lx106-elf-ld")
+set(CMAKE_C_LINK_EXECUTABLE "${CMAKE_C_LINKER} <OBJECTS> <CMAKE_C_LINK_FLAGS> <LINK_LIBRARIES> -o <TARGET>")
```
Comment (rasendubi) [inline gpiocontrol/source/esp8266/CMakeLists.txt:71]: Not sure, but these must be set in the toolchain file.

File: gpiocontrol/source/esp8266/CMakeLists.txt (hunk 1 above)
Comment (forGGe) [inline gpiocontrol/source/esp8266/CMakeLists.txt]: This must be addressed

File: gpiocontrol/source/esp8266/CMakeLists.txt (hunk 2 above)
Comment (forGGe) [inline gpiocontrol/source/esp8266/CMakeLists.txt:57]: This was discussed: has to be tested

File: gpiocontrol/source/esp8266/CMakeLists.txt (hunk 3 above)
Comment (forGGe) [inline gpiocontrol/source/esp8266/CMakeLists.txt]: This should be addressed

File: gpiocontrol/source/esp8266/CMakeLists.txt (hunk 4 above)
Comment (forGGe) [inline gpiocontrol/source/esp8266/CMakeLists.txt:66]: This should be addressed

File: gpiocontrol/source/esp8266/CMakeLists.txt (hunk 5 above)
Comment (forGGe) [inline gpiocontrol/source/esp8266/CMakeLists.txt:71]: This should be addressed in the scope of http://jira.kaaproject.org/browse/APP-88

File: gpiocontrol/source/esp8266/CMakeLists.txt (hunk 4 above)
Comment (Dubland) [inline gpiocontrol/source/esp8266/CMakeLists.txt:66]: `list(APPEND ...` inserts `;` instead of ` `

**Comment by forGGe** [comment]
Looks good, but should be reopened when appropriate develop branch will be created

**Comment by rasendubi** [comment]
@forGGe can't we just create develop branch for now?

**Comment by rasendubi** [comment]
Please, reopen against develop

**Pull request closed without merging**